File: src/BaseHmiTypes/Screens/ScreenManagement/HmiScreenWindowLayout.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.ScreenManagement;

public class HmiScreenWindowLayout : HmiScreenModelBase
{
    public HmiScreenWindowLayout()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiScreenWindowLayout;
    }

    public IList<HmiTopLevelScreenWindow> Windows { get; } = new List<HmiTopLevelScreenWindow>();
}

File: typescript/base-hmi-types/src/screens/screen-management/HmiScreenWindowLayout.ts
import { HmiScreenModelBase } from "../base/HmiScreenModelBase.js";
import { HmiTopLevelScreenWindow } from "../screen/HmiTopLevelScreenWindow.js";
import { HmiObjectType } from "../base/HmiObjectType.js";

export class HmiScreenWindowLayout extends HmiScreenModelBase {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiScreenWindowLayout;
  }

  readonly windows: HmiTopLevelScreenWindow[] = [];
}
